modules/macstadium_k8s_cluster/master.tf

```diff
@@ -0,0 +1,119 @@
+locals {
+  master_vm_name = "${var.name_prefix}-master"
+  master_ip      = "${vsphere_virtual_machine.master.clone.0.customize.0.network_interface.0.ipv4_address}"
+}
+
+resource "vsphere_virtual_machine" "master" {
+  name             = "${local.master_vm_name}"
+  folder           = "${var.folder}"
+  resource_pool_id = "${data.vsphere_compute_cluster.cluster.resource_pool_id}"
+  datastore_id     = "${data.vsphere_datastore.datastore.id}"
+
+  num_cpus  = 4
+  memory    = 4096
+  guest_id  = "${data.vsphere_virtual_machine.vanilla_template.guest_id}"
+  scsi_type = "${data.vsphere_virtual_machine.vanilla_template.scsi_type}"
+
+  disk {
+    label            = "disk0"
+    size             = "${data.vsphere_virtual_machine.vanilla_template.disks.0.size}"
+    eagerly_scrub    = "${data.vsphere_virtual_machine.vanilla_template.disks.0.eagerly_scrub}"
+    thin_provisioned = "${data.vsphere_virtual_machine.vanilla_template.disks.0.thin_provisioned}"
+  }
+
+  network_interface {
+    network_id = "${data.vsphere_network.internal.id}"
+  }
+
+  clone {
+    template_uuid = "${data.vsphere_virtual_machine.vanilla_template.id}"
+
+    customize {
+      network_interface {
+        ipv4_address = "${cidrhost("10.182.64.0/18", var.ip_base)}"
+        ipv4_netmask = 18
+      }
+
+      linux_options {
+        host_name = "${local.master_vm_name}"
+        domain    = "macstadium-us-se-1.travisci.net"
+      }
+
+      ipv4_gateway    = "10.182.64.1"
+      dns_server_list = ["1.1.1.1", "1.0.0.1"]
+      dns_suffix_list = ["vsphere.local"]
+    }
+  }
+
+  wait_for_guest_net_routable = false
+
+  connection {
+    host  = "${vsphere_virtual_machine.master.clone.0.customize.0.network_interface.0.ipv4_address}"
```

Comment:
what are all these `0`s  in the interpolation syntax? Attributes as `nil`?

Reply:
The clone and customize sections are represented as arrays, so you have to index into the first item.